### PR TITLE
Fix `registerBlockSingleProductTemplate` not respecting `isAvailableOnPostEditor` flag

### LINKED ISSUE #53604: `registerBlockSingleProductTemplate` doesn't respect `isAvailableOnPostEditor` flag
**Steps to reproduce:**
1. Create new post
2. Add Single Product block
3. Choose some product
4. Within Single Product try adding Product Image Gallery

**Actual:** Block is available and can be added even though it doesn't work on the frontend
**Expected:** Block is not available in the inserter on a post

**Notes:**

Most likely it's a side-effect of https://github.com/woocommerce/woocommerce/pull/53217.

### PR DESCRIPTION
**Pull request by kmanijak**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Due to changes in https://github.com/woocommerce/woocommerce/pull/53217, util didn't respect the `isAvailableOnPostEditor` flag and registered blocks in posts nevertheless.

Closes https://github.com/woocommerce/woocommerce/issues/53604

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

### Blocks - In posts

1. Create new post
2. Try inserting:
    - Product Price - should be unavailable (✖️)
    - Product Details - should be unavailable (✖️)
3. Insert Single Product block and choose some products
4. Within Single Product block try inserting:
    - Product Price - should be available (☑️)
    - Product Details - should be unavailable (✖️)

### Blocks - in Single Product Template

2. Go to Single Product template
2. Anywhere on the template, try inserting :
    - Product Price - should be available (☑️)
    - Product Details - should be available (☑️)

### Below steps are for developers only!

1. Allow Related Products in the inserter by replacing [this line](https://github.com/woocommerce/woocommerce/blob/b0bb52897a0bbe47a6866d5b5be9147a32bf986d/plugins/woocommerce-blocks/assets/js/blocks/product-query/variations/related-products.tsx#L143) with

```
scope: [ 'block', 'inserter' ],
```

In addition, pass the `isAvailableOnPostEditor: false,` flag in [line 146](https://github.com/woocommerce/woocommerce/blob/b0bb52897a0bbe47a6866d5b5be9147a32bf986d/plugins/woocommerce-blocks/assets/js/blocks/product-query/variations/related-products.tsx#L145).

#### Variation - disallowed in post

1. Create new post
2. Try inserting:
    - Related Products Controls - should be unavailable (✖️)
3. Insert Single Product block and choose some products
4. Within Single Product block try inserting:
    - Related Products Controls - should be unavailable (✖️)

#### Variation - allowed in post

0. Change the `isAvailableOnPostEditor` flag from `false` to `true`.
1. Create new post
2. Try inserting:
    - Related Products Controls - should be available (☑️)
3. Insert Single Product block and choose some products
4. Within Single Product block try inserting:
    - Related Products Controls - should be available (☑️)

> [!NOTE]  
> The last scenario is just a hypothetical and block itself will crash on the frontend as it's not meant to be used in this context. The key point is to check whether it appears or not in inserter based on the flag.
> Also, for some reason the block cannot be used as a direct child of Single Product block. This is also unrelated to this PR and like I said, block is not even meant to be used in that context hence it doesn't require investigation or fixing.

#### Variation - in Single Product Template

1. Go to Single Product template
2. Anywhere on the template, try inserting :
    - Related Products Controls - should be available (☑️)

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/tests/e2e/tests/related-products/related-products.block_theme.spec.ts [hunk 1]
```diff
@@ -17,10 +17,11 @@ const blockData: BlockData = {
...
 	test( "can't be added in the Post Editor", async ( { admin, editor } ) => {
 		await admin.createNewPost();
-		await editor.insertBlock( { name: blockData.slug } );
 		await expect(
-			await editor.getBlockByName( blockData.slug )
-		).toBeHidden();
+			editor.insertBlock( { name: blockData.slug } )
+		).rejects.toThrow(
+			new RegExp( `Block type '${ blockData.slug }' is not registered.` )
+		);
```
Comment (kmanijak) [inline plugins/woocommerce-blocks/tests/e2e/tests/related-products/related-products.block_theme.spec.ts:24]: Bring back test shape from before https://github.com/woocommerce/woocommerce/pull/53077/files#diff-11aa75b440fa37e554d1597dae0ffd7ebac524b0f597b8bb2ea123c13e64c7ff

File: plugins/woocommerce-blocks/tests/e2e/tests/single-product-details/single-product-details.block_theme.spec.ts [hunk 2]
```diff
@@ -18,10 +18,11 @@ test.describe( `${ blockData.slug } Block`, () => {
...
 	} ) => {
 		await admin.createNewPost();
-		await editor.insertBlock( { name: blockData.slug } );
 		await expect(
-			await editor.getBlockByName( blockData.slug )
-		).toBeHidden();
+			editor.insertBlock( { name: blockData.slug } )
+		).rejects.toThrow(
+			new RegExp( `Block type '${ blockData.slug }' is not registered.` )
+		);
```
Comment (kmanijak) [inline plugins/woocommerce-blocks/tests/e2e/tests/single-product-details/single-product-details.block_theme.spec.ts:25]: Bring back test shape from before https://github.com/woocommerce/woocommerce/pull/53077/files#diff-887d25d04962dca7c2b87d779d407db285297c95117319d74280a901294a3e7a

**Comment by gigitux** [comment]
Thanks for this fix! I tested your PR.

> Variation - allowed in post
Create new post
Try inserting:
Related Products Controls - should be unavailable (✖️)
Insert Single Product block and choose some products
Within Single Product block try inserting:
Related Products Controls - should be available (☑️)

In my test case, the Related Products control is available in the Post editor when not wrapped by the Single Product Block, but it is unavailable inside the Single Product Block.


image


Checking how the block is registered, I don't understand why this happens. I'm going to expect to see the block available in both context 🤔

**Comment by kmanijak** [comment]
> In my test case, the Related Products control is available in the Post editor when not wrapped by the Single Product Block, 

Ah! I made a mistake in testing steps, because variations can't have `ancestor` specified hence the block is available in the whole post context. Updating testing steps now to expect it's available outside and inside Single Product block.

---

> but it is unavailable inside the Single Product Block.

That's interesting, I see the block available. Unless you're using the "+" button? It seems there may be some bug upstream because I think it's the second time you cannot insert the block.

**Typing `/related products controls`**

image

**Using "+" button**

image

Interesting enough, it shows up when you click "Browse all":

image

**Comment by gigitux** [comment]
I don't think that it is a bug, when you open the sidebar, I guess that the area where you added the block is the root of the block hierarchy. 

For some reason, `Related Products Controls` can't be a child of the Single Product Block 🤔 


https://github.com/user-attachments/assets/a81cf134-aedc-4251-98d7-5f21273b7cf8

**Review by gigitux — APPROVED** [review]
Thanks for working on this! LGTM! 

As we discussed privately, the issue about the Related Products block and the Single Product Block is unrelated to this PR, so let's merge the PR updating the testing instructions!

**Comment by kmanijak** [comment]
> I don't think that it is a bug, when you open the sidebar, I guess that the area where you added the block is the root of the block hierarchy.
> For some reason, Related Products Controls can't be a child of the Single Product Block 🤔

I just checked and the same happens on `trunk`. But anyway, the Related Products Controls is not meant to be used in Single Product or specifically in post. And in fact the Related Product Controls crashes on frontend as it takes a wrong reference to display products from. 

What I mean is, in the testing step we force hypothetical scenario (the only variation using `registerBlockSingleProductTemplate ` to appear or not in posts based on flags) to verify the registration logic. And this is the only part that should, while it's not expected for the block itself to work correctly in that context.

**Pull request merged**